tools/findreplace.nim

```diff
@@ -0,0 +1,74 @@
+##[
+Experimental API, subject to change
+
+## Example usage
+using `ripgrep` for file listing:
+rg -l --color=auto -F 'defined(macosx)' lib | nim r tools/findreplace.nim --pattern:'defined\(macosx\)' --replacement:'defined(osx)' --dryrun:false
+
+## Design goal
+separation of concern between file listing and replacement, thanks to unix pipes,
```

Comment:
```
when defined(windows): {.warning: "Tool designed for Unix (Linux, BSD, OSX, etc)".}
```
So people dont try to run it on Windows?.

Reply:
why can't it work on windows?
all you need is to pass a list of files to stdin

Reply:
It explicitly says **Unix** pipes... :shrug: